tools/Ping1DMavlinkDriver.py

```diff
@@ -1,107 +1,137 @@
-ping1D_io.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
-ping1D_io.setblocking(False)
-
-## Send a request for distance_simple message to ping device
-def sendPing1DRequest():
-  data = PingMessage.PingMessage()
-  data.request_id = PingMessage.PING1D_DISTANCE_SIMPLE
-  data.src_device_id = 0
-  data.packMsgData()
-  ping1D_io.sendto(data.msgData, pingserver)
-
-# some extra information for the DISTANCE_SENSOR mavlink message fields
-min_distance = 20
-max_distance = 5000
-type = 2
-orientation = 25
-covarience = 0
-
-## Send distance_sensor message to autopilot
-def sendDistanceData(distance, deviceid, confidence):
-  print("sending distance %d confidence %d" % (distance, confidence))
-  if confidence < args.min_confidence:
-    distance = 0
-
-  autopilot_io.mav.distance_sensor_send(
-    (time.time() - tboot) * 1000, # time_boot_ms
-    min_distance, # min_distance
-    max_distance, # max_distance
-    distance/10, # distance
-    type, # type
-    deviceid, # device id
-    orientation,
-    covarience)
-
-while True:
-  time.sleep(0.001)
-  tnow = time.time()
-  
-  # request data from ping device
-  if(tnow > last_distance_measurement_time + ping_interval_ms):
-    if(tnow > last_ping_request_time + ping_interval_ms):
-      last_ping_request_time = time.time()
-      sendPing1DRequest()
-
-  # read data in from ping device
-  try:
-    data, address = ping1D_io.recvfrom(4096)
-  except Exception as e:
-            if e.errno == errno.EAGAIN:
-              pass # waiting for data
-            else:
-              raise e
-            continue # back to top of loop
-
-  # decode data from ping device, forward to autopilot
-  for byte in data:
-    if ping_parser.parseByte(byte) == PingMessage.PingParser.NEW_MESSAGE:
-      if ping_parser.rxMsg.message_id in distance_messages:
-        last_distance_measurement_time = time.time()
-        distance = ping_parser.rxMsg.distance
-        deviceid = ping_parser.rxMsg.src_device_id
-        confidence = ping_parser.rxMsg.confidence
-        sendDistanceData(distance, deviceid, confidence)
+PARSER = argparse.ArgumentParser(description="Ping1D to mavlink bridge.")
+PARSER.add_argument('--ping',
+                    action="store",
+                    type=str,
+                    default="0.0.0.0:9090",
+                    help="Ping device udp address and port. ex \"0.0.0.0:9090\""
+                    )
+PARSER.add_argument('--mavlink',
+                    action="store",
+                    type=str,
+                    default="0.0.0.0:9000",
+                    help="Mavlink udp address and port. ex \"0.0.0.0:9000\""
+                    )
+PARSER.add_argument('--min-confidence',
+                    action="store",
+                    type=int,
+                    default=50,
+                    help="Minimum acceptable confidence percentage for depth measurements.\""
+                    )
+ARGS = PARSER.parse_args()
+
+def main():
```

Comment:
This isn't a module, it's a script (always need to run main, or there's no point), so why go through this __main__ thing?

Reply:
@Williangalvani

Reply:
tricky one...
If you ever need to import it as a module (for testing, for example), then it should be using functions, otherwise doesn't seem to make a difference.

Reply:
But for that, should I capitalize all variables ?

Reply:
No, why?
AFAIR you only capitalize class names.

Reply:
`Constant name "abc" doesn't conform to UPPER_CASE naming style (invalid-name)
`

Reply:
Just do
```
abc = 2
abc += 1
```
with pylint

Reply:
is this "abc" a constant?

Reply:
Ok. From this conversation, my vote is for the main function because it will work in any case. I'd be interested in seeing some best practice on the matter anyway though if you guys come across some.